### DISCUSSION TITLE
directive order does not work.

### DISCUSSION THREAD

**Discussion by edwardLz** [supabase/supabase#1169] [General]
Hello, I have the following problem, the truth is I do not know if I am implementing this method wrong.
I try to apply a command to my query but it is not reflected in the result.

This is one of the implementations that I have, I did it the same as it says in the documentation, but it does not work for me, any idea why?

```js
supabase
    .from(table)
    .select("*, course:courseId (*, level:levelId(*))", {
      count: "exact",
    })
    .eq("course.schoolId", schoolId)
    .is("deletedAt", null)
    .order("letter", {foreignTable: "course", ascending: true})
    .range(from, to)
```

**Comment by steve-chavez** [comment]
> but it does not work for me

What's the error message?

**Reply by edwardLz** [reply]
it doesn't throw an error, it just doesn't sort the results.
I also tried this way:
`.order("course.letter.asc")`

I found this last one in some answers on this forum (#229), but it doesn't work either and in this case if it throws an error:
`details: "unexpected "l" expecting "asc", "desc", "nullsfirst" or "nullslast""
message: ""failed to parse order (course.letter.asc.asc.nullslast)" (line 1, column 8)"`
But if I try to do the order as the doc says it just doesn't do anything

**Reply by kiwicopple** [reply]
That way is quite old syntax, so you should definitely use the syntax in Steve's answer (https://github.com/supabase/supabase/discussions/229#discussioncomment-89979) 

Could you try to trim you example a bit just to isolate the root cause? I don't know what `table` is in your example, but assuming it is something like `'students'`, then perhaps:

```js
supabase
    .from('students')
    .select('id, name, course ( name, letter )')
    .order('letter"' { foreignTable: "course", ascending: true })
```

If that works, build up from there.

I can't really see anything wrong with your query above, but our tests are passing so it doesn't look like any regression from our side

**Reply by edwardLz** [reply]
Ok !, I still do not solve it, simplify the queries as much as possible and it does not order, and if I do it like Steve #229, it gives me the previous error
`supabase
    .from("students")
    .select("*")
    .order("id.asc")`

**Reply by steve-chavez** [reply]
@edwardLz Try it like:

```js
supabase.from("students").select("*").order("id", {ascending: true})
```

**Reply by edwardLz** [reply]
@steve-chavez, that works without problems, I start, but what does not work is when I want to sort by an attribute of a foreign table, I show the schema of my tables, maybe there is something that I do not see in the relationship, and that is affecting the query

![image](https://user-images.githubusercontent.com/46491893/115644441-96263500-a2ec-11eb-908d-8eae56622ddb.png)

that's the relationship of the three tables, what I'm trying to do is sort by level aliases and then by course letter, and that's what doesn't work.

my original query is this:
```
supabase
.from("students")
.select("*, course:courseId (*, level:levelId(*))")
.order("alias", {foreignTable: "level", ascending: true})
.order("letter", {foreignTable: "course", ascending: true})
```

**Reply by steve-chavez** [reply]
Hm, that's strange because according to this [test](https://github.com/supabase/postgrest-js/blob/master/test/resource-embedding.ts#L43-L50) that should work. Perhaps you can share some sample results and report the bug on https://github.com/supabase/postgrest-js/?

**Reply by edwardLz** [reply]
and where in the documentation is what Steve proposes in #229

**Reply by jamorajj** [reply]
Hi @edwardLz , did you manage to have this working? I'm currently experiencing this as well;

    supabase
    .from('inventory')
    .select(
      id,
      quantity,
      quantity_warning,
      expiration_date,
      product_id ( name, sku, brand, category )
    )
    .eq('branch_id', id)
    .order('name', { ascending: true, foreignTable: 'product' });

my result yields an unsorted list in terms of product name.

Thanks!

**Reply by steve-chavez** [reply]
> my result yields an unsorted list in terms of product name.

@jamorajj  Are you expecting the inventory rows to be ordered by product name?

> .order('name', { ascending: true, foreignTable: 'product' });

What the above does is to order the product rows, it doesn't affect the parent(inventory) rows.

**Reply by jamorajj** [reply]
@steve-chavez -- yes, that's my expected result. is there a way that I could achieve the inventory rows to be sorted by product name? thanks for looking into this :)

**Comment by Dizotoff** [comment]
Experience the same issue here
```
 await supabase
              .from("markets")
              .select("companies(*)")
              .order("company_name", { foreignTable: "companies", ascending: false });
```
Returns an unsorted list. I think the snapshot test mentioned above might be broken.

**Reply by steve-chavez** [reply]
I think you might be running into this issue https://github.com/supabase/postgrest-js/issues/198
